### PR TITLE
update DEVELOPER docs

### PR DESCRIPTION
**Pull request by arkodg**
Fixes: https://github.com/envoyproxy/gateway/issues/295
Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -26,8 +26,38 @@ Envoy Gateway is built using a [make][make]-based build system. Our CI is based
...
+* Run `make test` to run the golang tests.
+
+### Running code linters
+* Run `make lint` to make sure your code passes all the linter checks.
+
+### Building and Pushing the Image
+* Run `IMAGE=foo/bar make image` to build the docker image.
+* Run `IMAGE=foo/bar make push-multiarch` to build and push the multi-arch docker image.
+
+**_NOTE:_**  Replace `IMAGE` with your registry's image name.
```
Comment (danehans) [inline DEVELOPER.md:44]: Can you be more specific with this note, and/or provide an example above? `IMAGE` should be `REGISTRY/REPOSITORY/IMAGE_NAME`. For example:
```
IMAGE=docker.io/you/gateway-dev make push-multiarch
```

File: DEVELOPER.md [hunk 2]
```diff
@@ -26,8 +26,38 @@ Envoy Gateway is built using a [make][make]-based build system. Our CI is based
...
+
+### Deploying Envoy Gateway in Kubernetes
```
Comment (danehans) [inline DEVELOPER.md:46]: I think most devs will want to use `make create-cluster` to dev against a Kind cluster. If you agree, can you add a step for this target and say that you can build/push a local EG image to the cluster with `make kube-install-image`?

File: DEVELOPER.md [hunk 3]
```diff
@@ -26,8 +26,42 @@ Envoy Gateway is built using a [make][make]-based build system. Our CI is based
   separately.
 
 ## Quick start
+* Run `make help` to see all the available targets to build, test and run `envoy-gateway`.
```
Comment (danehans) [inline DEVELOPER.md:29]: @arkodg one last nit, do you mind reviewing all the bullets?

Comment (arkodg) [reply DEVELOPER.md:29]: this page already has bullets, prefer keeping the styling the same, similar to the rest of the docs such as [design](https://github.com/envoyproxy/gateway/blob/main/docs/design/SYSTEM_DESIGN.md) docs in this repo

**Review by danehans — APPROVED** [review]

**Pull request merged**